homeassistant/components/smartthings/__init__.py

```diff
@@ -127,7 +127,9 @@
                 )
                 devices.remove(device)
 
-        await asyncio.gather(*(retrieve_device_status(d) for d in devices.copy()))
+        await safe_wait(
+            (retrieve_device_status(d) for d in devices.copy()), logger=_LOGGER
```

Comment:
`retrieve_device_status` already handles exceptions and we want it to apply this function to all of the devices. 
 I think the usage of `gather` was correct in this case.

Reply:
maybe and maybe on other parts of the code. But if we replace all wait and gather on our codebase, it's easier for user/review to say, use that function.